### PR TITLE
Github activity in stream

### LINKED ISSUE #136: frontend github
github will be frontend only and it needs to be integrated to stream

### PR DESCRIPTION
**Pull request by derrick-w**
resolves #136 

Activity of users you follow on Github now appears in stream
Github rate limit workaround not implemented (Etag)

Screenshot_20190330-1

### CODE REVIEW

File: frontend/src/components/StreamPost.jsx [hunk 1]
```diff
@@ -320,6 +333,7 @@ class StreamPost extends Component {
 	}
 }
 
+
 StreamPost.propTypes = {
```
Comment (shchau) [inline frontend/src/components/StreamPost.jsx:337]: Add isGithub as an non-required proptype. :+1:

File: frontend/src/pages/Author.jsx [hunk 2]
```diff
@@ -398,6 +397,11 @@ class Author extends Component {
         );
     }
 
+    getFullAuthorIdFromURL(path) {
```
Comment (hiufungk) [inline frontend/src/pages/Author.jsx:400]: why is this unused function added？

File: frontend/src/pages/Stream.jsx [hunk 3]
```diff
@@ -3,15 +3,57 @@ import StreamFeed from '../components/StreamFeed';
...
+	constructor(props) {
+		super(props);
+		this.state = {
+			github: '',
+		};	
+		this.fetchProfile = this.fetchProfile.bind(this);
+	};	
 	
+	fetchProfile() {
+		//todo deal with other hosts
```
Comment (hiufungk) [inline frontend/src/pages/Stream.jsx]: why need to deal with other host?

File: frontend/src/pages/Stream.jsx [hunk 4]
```diff
@@ -3,15 +3,57 @@ import StreamFeed from '../components/StreamFeed';
...
+		const hostUrl = "/api/author/"+ utils.getShortAuthorId(Cookies.get("userID")),
```
Comment (hiufungk) [inline frontend/src/pages/Stream.jsx]: value from store || cookies.get instead of only cookie

**Review by hiufungk — CHANGES_REQUESTED** [review]

**Review by hiufungk — APPROVED** [review]

**Review by shchau — APPROVED** [review]

**Review by forgeno — APPROVED** [review]

**Pull request merged**